### ISSUE TITLE
Fixed overindented enrollment configuration of the Windows agent

### ISSUE THREAD

**Issue by davidcr01** [wazuh/wazuh-qa#4095]
| Target version | Related issue | Related PR |
|--------------------|--------------------|-----------------|
|        4.5                |        https://github.com/wazuh/wazuh/issues/15324                |       https://github.com/wazuh/wazuh/pull/16720             |

## Description

The PR proposes to fix the overindented Widows agent enrollment configuration. 
The current configuration looks like this:
```
    <enrollment>
      	<enabled>yes</enabled>
        <manager_address>10.0.0.2</manager_address>
    </enrollment>
```

With the fix, the configuration block now looks like this:
```
    <enrollment>
      <enabled>yes</enabled>
      <manager_address>10.0.0.2</manager_address>
    </enrollment>
```

## Proposed checks

- [x] [Generate a new package](https://documentation.wazuh.com/current/development/packaging/generate-windows-package.html).
- [x] Test installation of the new package.
- [x] Test upgrade of the new package.

## Steps to reproduce

- Download the Wazuh Windows agent package
- Install Windows agent using enrollment deployment variables, for example: `./wazuh-agent-4.5.0-1.msi /q WAZUH_REGISTRATION_SERVER="10.0.0.2"`
- Check enrollment configuration
```
    <enrollment>
      	<enabled>yes</enabled>
        <manager_address>10.0.0.2</manager_address>
    </enrollment>
```

## Expected results

The enrollment configuration should look like this:
```
    <enrollment>
      <enabled>yes</enabled>
      <manager_address>10.0.0.2</manager_address>
    </enrollment>
```

## Configuration and considerations

When installing the Wazuh agent in the Windows host, make sure that the terminal has been run as Administrator.

**Comment by pro-akim** [comment]
| Wazuh-Manager OS| Wazuh-Manager Version | Agent OS | Agent Version |
|--------------------|--------------------|-----------------|-----------------|
| Centos8| 4.5.0-0.commitb82bc29.x86_64 | Windows Server  |4.4.1-1  & 4.5.0-0.commitb82bc29 |
Having  systemctl firewalld disabled


Three tests were performed for this issue.

**1. Check wazuh-agent 4.4.1-1 enrollment and its function**

Install wazuh-agent-4.4.1-1.msi in windows VM and check 'C:\Program Files (x86)\ossec.conf' indentation:

```
    <enrollment>
        <enabled>yes</enabled>
        <manager_address>172.16.1.207</manager_address>
    </enrollment>
```

Configure the server address:

```
    <server>
      <address>172.16.1.207</address>
      <port>1514</port>
      <protocol>tcp</protocol>
    </server>
```

Start the agent:

```
NET START WazuhSvc
```

Check in wazuh-manager:

```
/var/ossec/bin/manage_agents -l
```

Result:

```
Available agents: 
   ID: 001, Name: windows-agent, IP: any
```

**2. Check wazuh-agent-4.5.0-0.commitb82bc29 enrollment and its function**

Install wazuh-agent-4.5.0-0.commitb82bc29.msi in windows VM and check 'C:\Program Files (x86)\ossec.conf' indentation:

  ```
    <enrollment>
      <enabled>yes</enabled>
      <manager_address>172.16.1.207</manager_address>
    </enrollment>
```

Configure the server address:

```
    <server>
      <address>172.16.1.207</address>
      <port>1514</port>
      <protocol>tcp</protocol>
    </server>
```

Start the agent:
```
NET START WazuhSvc
```

Check in wazuh-manager:

```
/var/ossec/bin/manage_agents -l
```

Result:

```
Available agents: 
   ID: 002, Name: windows-agent, IP: any
```

**3. Update to wazuh-agent-4.5.0-0.commitb82bc29 and check enrollment and its function**

Install wazuh-agent-4.4.1-1.msi in windows, update with wazuh-agent-4.5.0-0.commitb82bc29.msi in windows VM and check 'C:\Program Files (x86)\ossec.conf' indentation:

  ```
    <enrollment>
        <enabled>yes</enabled>
        <manager_address>172.16.1.207</manager_address>
    </enrollment>
```

Configure the server address:

```
    <server>
      <address>172.16.1.207</address>
      <port>1514</port>
      <protocol>tcp</protocol>
    </server>
```

Start the agent:

```
NET START WazuhSvc
```

Check in wazuh-manager:

```
/var/ossec/bin/manage_agents -l
```

Result:

```
Available agents: 
   ID: 003, Name: windows-agent, IP: any
```

In conclusion:

    Version 4.5.0-0 of Wazuh-agent has successfully fixed the super indentation issue.
    Updating the version does not modify ossec.conf.
    Wazuh-manager version 4.5.0-0 is compatible with both Wazuh-agent version 4.5.0-0 and Wazuh-agent version 4.4.1-1.

**Comment by pro-akim** [comment]
## Review data

| Tester | Related PR |
|--------|-------------------------|
| @pro-akim   | [16720](https://github.com/wazuh/wazuh/pull/16720)   |

### Testing environment

| OS | OS version | Deployment                                    | Image/AMI | Notes |
|----|------------|-----------------------------------------------|-----------|-------|
|  CentOS  | 8 | Vagrant | generic/centos_8 |


### Tested packages

Manager was built from sources since Jenkins is down

**Comment by pro-akim** [comment]
## Testing results

### 
<details>
<summary><b>Case 1:</b> Install <code>wazuh-agent-4.4.1-1</code> and check enrollment and its function :green_circle:</summary>
<br>

1. Install <code>wazuh-agent-4.4.1-1</code> in windows:

```C:\Program Files (x86)\ossec.conf
    <enrollment>
        <enabled>yes</enabled>
        <manager_address>172.16.1.207</manager_address>
    </enrollment>
```
2. Configure the server address:

```C:\Program Files (x86)\ossec.conf
    <server>
      <address>172.16.1.207</address>
      <port>1514</port>
      <protocol>tcp</protocol>
    </server>
```
3. Start the agent:

```cmd
NET START WazuhSvc
```

4. Check in wazuh-manager:

```bash
/var/ossec/bin/agent_control -l
```

5. Result:

```bash
Wazuh agent_control. List of available agents:
   ID: 000, Name:qaHost6 (server), IP: 127.0.0.1, Active/local
   ID: 001, Name: windows-agent, IP: any, Active
```

</details>
<details>
<summary><b>Case 2:</b> Install <code>wazuh-agent-4.5.0-0.commitb82bc29</code> and check enrollment and its function :green_circle:</summary>
<br>

1. Install <code>wazuh-agent-4.5.0-0.commitb82bc29</code> in windows:

```C:\Program Files (x86)\ossec.conf
    <enrollment>
      <enabled>yes</enabled>
      <manager_address>172.16.1.207</manager_address>
    </enrollment>
```
2. Configure the server address:

```C:\Program Files (x86)\ossec.conf
    <server>
      <address>172.16.1.207</address>
      <port>1514</port>
      <protocol>tcp</protocol>
    </server>
```
3. Start the agent:

```cmd
NET START WazuhSvc
```

4. Check in wazuh-manager:

```bash
/var/ossec/bin/agent_control -l
```

5. Result:

```bash
Wazuh agent_control. List of available agents:
   ID: 000, Name:qaHost6 (server), IP: 127.0.0.1, Active/local
   ID: 002, Name: windows-agent, IP: any, Active
```

</details>

<details>
<summary><b>Case 3:</b> Update to <code>wazuh-agent-4.5.0-0.commitb82bc29</code> and check enrollment and its function :green_circle:</summary>
<br>

1. Install <code>wazuh-agent-4.4.1-1.msi</code> in windows:

```C:\Program Files (x86)\ossec.conf
    <enrollment>
        <enabled>yes</enabled>
        <manager_address>172.16.1.207</manager_address>
    </enrollment>
```
2. Configure the server address:

```C:\Program Files (x86)\ossec.conf
    <server>
      <address>172.16.1.207</address>
      <port>1514</port>
      <protocol>tcp</protocol>
    </server>
```
3. Start the agent:

```cmd
NET START WazuhSvc
```

4. Check in wazuh-manager:

```bash
/var/ossec/bin/agent_control -l
```

5. Result:

```bash
Wazuh agent_control. List of available agents:
   ID: 000, Name:qaHost6 (server), IP: 127.0.0.1, Active/local
   ID: 003, Name: windows-agent, IP: any, Active
```

6. Stop the agent:

```cmd
NET STOP WazuhSvc
```

7. Check in wazuh-manager:

```#bash
/var/ossec/bin/agent_control -l
```

8. Result:

```bash
Wazuh agent_control. List of available agents:
   ID: 000, Name:qaHost6 (server), IP: 127.0.0.1, Active/local
   ID: 003, Name: windows-agent, IP: any, Disconnected
```

9. Update to <code>wazuh-agent-4.5.0-0.commitb82bc29</code> in windows:

```#C:\Program Files (x86)\ossec.conf
    <enrollment>
        <enabled>yes</enabled>
        <manager_address>172.16.1.207</manager_address>
    </enrollment>
```
10. Restart the agent:

```cmd
NET START WazuhSvc
```

9. Check in wazuh-manager:

```#bash
/var/ossec/bin/agent_control -l
```

11. Result:

```bash
Wazuh agent_control. List of available agents:
   ID: 000, Name:qaHost6 (server), IP: 127.0.0.1, Active/local
   ID: 003, Name: windows-agent, IP: any, Active
```
</details>

**Comment by juliamagan** [comment]
## Closing conclusion 👍🏼  

Everything seems to be working properly
